liblangutil/ErrorReporter.h

```diff
@@ -87,6 +87,11 @@ class ErrorReporter
 
 	void typeError(SourceLocation const& _location, std::string const& _description);
 
+	void typeError(
+		SourceLocation const& _location,
+		std::initializer_list<std::string const> const& _descriptions
```

Comment:
Why is this necessary?

Reply:
Otherwise instead of doing

```
		m_errorReporter.typeError(
			arguments.front()->location(),
			{"Invalid type for argument in function call. "
			"Invalid implicit conversion from " +
			type(*arguments.front())->toString() +
			" to bytes memory requested.",
			details.message()}
		);
```
I would need to do

```
		m_errorReporter.typeError(
			arguments.front()->location(),
			"Invalid type for argument in function call. "
			"Invalid implicit conversion from " +
			type(*arguments.front())->toString() +
			" to bytes memory requested." +
			(details.message().empty() ? "" : " " + details.message())
		);
```

for every single occurrence which imo looks rather messy and not easy to read.
This approach seemed the cleanest way to me.

Reply:
If we decide to do it like that, I would actually prefer a template function with variable arguments instead of an initializer list - the `{` look a bit weird in the argument list.

Reply:
I opted for the initializer list because it provides type safety and.. well templates always cost compile time, too.

Reply:
I'd also vote for implementing it with a variadic template. The initializer list does indeed look a bit off.

Reply:
@Marenz if this clashes with something, you can always add another function with a different name.

Reply:
Well, that would be an easy way to work around my current problem.